FireRender.Maya.Src/FireRenderVolume.cpp

```diff
@@ -441,19 +441,16 @@ bool NorthstarRPRVolume::TranslateVolume()
 			RPR_GRID_INDICES_TOPOLOGY_XYZ_U32
 		);
 
+		MFnDependencyNode depNode(node);
+		float maxDensityValue = RPRVolumeAttributes::GetDensityMultiplier(depNode) * 1.1f;
+
 		auto densityGridNode = frw::GridNode(context()->GetMaterialSystem());
 		densityGridNode.SetGrid(m_densityGrid);
-
-		auto densityLookupNode = CreateLookupTextureNode(this, vdata.densityGrid.valuesLookUpTable, densityGridNode);
-
 		volumeShader.xSetValue(RPR_MATERIAL_INPUT_DENSITYGRID, densityGridNode);
-
-		//now density is a constant value because density lookup is temporary unavailable in Northstar
-		//volumeShader.xSetValue(RPR_MATERIAL_INPUT_DENSITY, densityLookupNode);
-		volumeShader.xSetParameterF(RPR_MATERIAL_INPUT_DENSITY, 1000.0f, 1.0f, 1.0f, 1.0f); 
+		volumeShader.xSetParameterF(RPR_MATERIAL_INPUT_DENSITY, maxDensityValue, 1.0f, 1.0f, 1.0f);
 	}
 	
-	if (vdata.albedoGrid.IsValid()) // grid exists
+	if (vdata.albedoGrid.IsValid() && vdata.albedoGrid.gridOnIndices.size() > 0) // grid exists
```

Comment:
Please consider moving check gridOnIndices.size() > 0 to IsValid() method.

If I understood correctly grid with gridOnIndices.size() == 0 would be invalid, right?